src/thor/route_action.cc

```diff
@@ -424,7 +424,8 @@ void thor_worker_t::path_arrive_by(Api& api, const std::string& costing) {
 
   auto route_two_locations = [&, this](auto& origin, auto& destination) -> bool {
     // Get the algorithm type for this location pair
-    thor::PathAlgorithm* path_algorithm = get_path_algorithm(costing, *origin, *destination, options);
+    thor::PathAlgorithm* path_algorithm =
+        this->get_path_algorithm(costing, *origin, *destination, options);
```

Comment:
can you share the compiler messages that are making you use this syntax, is it something about being ambiguous? we dont use `this` to access members anywhere else in the code base that im aware of so itd be sad to start doing that stylistically if we can instead just rename some other methods that the compiler is complaining about?

Reply:
It was smth along the lines of `Missing object` on those lines where `this->` was missing. It's not a general access member thing, rather smth to do with capturing `this` inside the lambda. I can reproduce tmrw and paste the trace if you want.

Reply:
here it is:

```make
/valhalla/src/thor/route_action.cc: In instantiation of ‘valhalla::thor::thor_worker_t::path_arrive_by(valhalla::Api&, const string&)::<lambda(auto:2&, auto:3&)> [with auto:2 = std::reverse_iterator<google::protobuf::internal::RepeatedPtrIterator<valhalla::Location> >; auto:3 = std::reverse_iterator<google::protobuf::internal::RepeatedPtrIterator<valhalla::Location> >]’:
/valhalla/src/thor/route_action.cc:518:49:   required from here
/valhalla/src/thor/route_action.cc:441:87: error: cannot call member function ‘std::vector<std::vector<valhalla::thor::PathInfo> > valhalla::thor::thor_worker_t::get_path(valhalla::thor::PathAlgorithm*, valhalla::Location&, valhalla::Location&, const string&, const valhalla::Options&)’ without object
     auto temp_paths = get_path(path_algorithm, *origin, *destination, costing, options);
                                                                                       ^
/valhalla/src/thor/route_action.cc:445:5: error: unable to deduce ‘auto&&’ from ‘temp_paths’
     for (auto& temp_path : temp_paths) {
     ^~~
```

Reply:
In a quick search I couldn't find any other similar error report, so I'd blame the old GCC?! In any case, it seems capturing `[&, this]` is redundant and could be simply `[&]`, which compiles both without the above fix on a recent GCC (v10.2) and with the above fix on v6.3.0.

Reply:
ah man that does suck, very unfortunate old compiler bug

Reply:
jeez, you're up early on a sat ;)

Reply:
ah DST hit in..